waf.go

```diff
@@ -13,15 +13,15 @@ import (
 	"github.com/hashicorp/go-multierror"
 )
 
-// UnsupportedTargetError is a wrapper error type helping to handle the error
+// WafDisabledError is a wrapper error type helping to handle the error
 // case of trying to execute this package on an unsupported target environment.
-type UnsupportedTargetError struct {
+type WafDisabledError struct {
```

Comment:
We are losing information (and breaking the public API too):
1. `UnsupportedTargetError` can be unintended and could deserve its own error-management logic
2. `WafDisabledError` is rather more likely coming from an explicit action.

I'd suggest we keep it open with two error types, that the caller can type-assert or error.Unwrap if needed
.

Reply:
`SupportsTarget()` becomes ambigous too for the disabled case, so we should work the build tags so that the unsupported target is higher prio than the disabled case

Reply:
I suggest you review dd-trace-go's and datadog-agent's use of `Health()` and `SupportsTarget()` to check we are good and we won't fall into different/new conditions.

Reply:
I managed to change thing into something I am comfortable with